lib/data_update_scripts/20210218041143_backfill_usernames.rb

```diff
@@ -1,9 +1,10 @@
 module DataUpdateScripts
   class BackfillUsernames
     def run
-      User.where(username: nil).find_each do |user|
-        user.update(username: "user#{user.id}")
-      end
+      # Now that we have a DB constraint "username IS NOT NULL" this script will always fail
+      # User.where(username: nil).find_each do |user|
+      #   user.update(username: "user#{user.id}")
+      # end
```

Comment:
By "will always fail", do you mean it will never find anything or it actually raises an exception?

Reply:
It raised an exception in the tests because the only way to test this behavior is to set a `username` to nil but you can't do that anymore now that the constraint is enforced, thus there's no possibility that this will find anything as well... :D